### PR TITLE
[FIXED JENKINS-28706] - Hide Docker Traceability Root Action by default

### PR DESCRIPTION
**Pull request by oleg-nenashev**
On the top of #11 

@reviewbybees

### CODE REVIEW

**Comment by tfennelly** [comment]
:+1:

File: README.md [hunk 1]
```diff
@@ -27,8 +27,9 @@ This [Jenkins](http://jenkins-ci.org) plugin allows the tracking of the creation
       Web interfaces are being managed by common Jenkins Read permission
  * **Submit** - Allows the submission deployment records from the [remote API](#API)
  * **Delete** - Allows the deletion deployment records or entire fingerprints
-
-Currently, the plugin does not require additional configuration.
+4. Setup the Docker image fingerprint creation mode
```
Comment (jtnord) [inline README.md]: not related to this change?

Comment (oleg-nenashev) [inline README.md]: It's from #11

File: src/main/java/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPlugin.java [hunk 2]
```diff
@@ -55,10 +60,54 @@
...
+        }
+    }
+
+    void configure(DockerTraceabilityPluginConfiguration configuration) throws IOException {
+        this.configuration = configuration;
+        save();
+    }
+    
+    @Override
+    public void configure(StaplerRequest req, JSONObject formData) throws IOException, ServletException, Descriptor.FormException {
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPlugin.java]: I thought this needed to [return a boolean](http://javadoc.jenkins-ci.org/hudson/model/Descriptor.html#configure(org.kohsuke.stapler.StaplerRequest, net.sf.json.JSONObject))?

Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPlugin.java]: Oh.. plugin not descriptor.

**Comment by jtnord** [comment]
:+1: we must be using github wrong if we can't separate PRs built on top of PRs correctly... (or the tooling sucks)

**Comment by jglick** [comment]
> if we can't separate PRs built on top of PRs correctly

A PR on top of another PR will show the cumulative diff: what you will actually get in `master` if you ask to merge it.

Generally you should review the upstream PR first, so it can get merged. Then if the author of the downstream PR does a `git merge master && git push` the cumulative diff will correctly show only that PR’s changes.

File: src/main/java/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPluginConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import hudson.model.Describable;
+import hudson.model.Descriptor;
+import javax.annotation.Nonnull;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * Configuration of {@link DockerTraceabilityPlugin}.
+ * @author Oleg Nenashev
+ */
+public class DockerTraceabilityPluginConfiguration implements Describable<DockerTraceabilityPluginConfiguration> {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPluginConfiguration.java]: Use `GlobalConfiguration`.

File: src/main/resources/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPluginConfiguration/config.properties [hunk 4]
```diff
@@ -0,0 +1,2 @@
+createImageFingerprints.title=Create image fingerprints on-demand
+showRootAction.title=Show Docker Traceability action on the main side panel
```
Comment (jglick) [inline src/main/resources/org/jenkinsci/plugins/docker/traceability/DockerTraceabilityPluginConfiguration/config.properties]: Better than nothing, but still a very weird thing to be asking users to configure. We need to either find a way to determine mechanically whether this functionality is needed; or come up a with general UI landing site for “environmental information” that does not clog the main sidepanel (CC @gusreiber).

**Pull request merged**